### PR TITLE
Version 1.16.1

### PR DESCRIPTION
**Pull request by SC426**
jupytext 1.16.1

**Destination channel:** defaults

### Links

- [PKG-4159](https://anaconda.atlassian.net/browse/PKG-4159)
- [Upstream repository](https://github.com/mwouts/jupytext/tree/v1.16.1)
- [Upstream changelog/diff](https://github.com/mwouts/jupytext/blob/v1.16.1/CHANGELOG.md)

### Explanation of changes:

- Update version
- Clean up scripts and entry points
- Update dependencies

[PKG-4159]: https://anaconda.atlassian.net/browse/PKG-4159?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,40 +1,37 @@
...
-    - setuptools
-    - wheel
+    - hatchling >=1.5.0
+    - hatch-jupyter-builder >=0.5
+    - jupyterlab >=4
   run:
     - python
     - nbformat
+    - mdit-py-plugins
+    - markdown-it-py >=1.0.0
```
Comment (Jrice1317) [inline recipe/meta.yaml]:
```suggestion
    - markdown-it-py >=1.0
```
I don't think this change is that important, but just thought it be good to match upstream.

Comment (Jrice1317) [reply recipe/meta.yaml]: https://github.com/mwouts/jupytext/blob/v1.16.1/pyproject.toml#L36

File: recipe/meta.yaml [hunk 2]
```diff
@@ -44,9 +41,7 @@ test:
   commands:
     - pip check
     - jupytext --help
-    - test -f ${PREFIX}/share/jupyter/nbextensions/jupytext/index.js                                              # [unix]
     - test -f ${PREFIX}/share/jupyter/labextensions/jupyterlab-jupytext/package.json                              # [unix]
```
Comment (Jrice1317) [inline recipe/meta.yaml:43]: We may be able to remove the other two similar commands as well, as I can not find how they are used.

It seems we could use pytest-cov and I also see [this](https://github.com/mwouts/jupytext/blob/v1.16.1/pyproject.toml#L155), which seems slightly related but it still makes me believe they can be removed at first glance, but I'm interested to hear yours and others thoughts on this.

Comment (SC426) [reply recipe/meta.yaml:43]: I believe it's testing to see if this file exists https://github.com/mwouts/jupytext/blob/v1.16.1/jupyterlab/package.json

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,40 +1,37 @@
...
     - pip
-    - python
-    - nodejs 18.16.0
-    - jupyter-packaging
-    - jupyterlab >=3.0,<4.0
-    - setuptools
-    - wheel
+    - hatchling >=1.5.0
+    - hatch-jupyter-builder >=0.5
+    - jupyterlab >=4
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]: I'm not sure if these two are necessary. The extension files seem to already be in the sdist...

File: recipe/meta.yaml (hunk 2 above)
Comment (Jrice1317) [reply recipe/meta.yaml:43]: Sorry, I tagged the wrong line. 🤦‍♀️ I meant [this line](https://github.com/mwouts/jupytext/blob/v1.16.1/pyproject.toml#L150)

File: recipe/meta.yaml (hunk 3 above)
Comment (SC426) [reply recipe/meta.yaml]: I don't think `jupyterlab` is necessary, but it seems `jupyter-builder` is.
 
You get `hatchling.plugin.exceptions.UnknownPluginError: Unknown build hook: jupyter-builder` without it.

Comment (SC426) [reply recipe/meta.yaml]: https://github.com/mwouts/jupytext/blob/1ef1dea7d1d63e418ecc093025510e7b941ade9c/pyproject.toml#L140

File: recipe/meta.yaml (hunk 2 above)
Comment (SC426) [reply recipe/meta.yaml:43]: My only opinion is that some tests are better than none, and this is the way conda forge is doing it.

**Review by JamesRobertsonGames — APPROVED** [review]
Looks good to me 👍

**Review by cbouss — APPROVED** [review]

**Pull request merged**